naptha_sdk/client/hub.py

```diff
@@ -243,70 +246,82 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
 
 async def user_setup_flow(hub_url, public_key):
     async with Hub(hub_url, public_key) as hub:
-        username, password = os.getenv("HUB_USER"), os.getenv("HUB_PASS")
-        username_exists, password_exists = len(username) > 1, len(password) > 1
-        public_key = get_public_key(os.getenv("PRIVATE_KEY")) if os.getenv("PRIVATE_KEY") else None
-        logger.info(f"Checking if user exists... User: {username}")
-        user = await hub.get_user_by_username(username)
-        user_public_key = await hub.get_user_by_public_key(public_key)
-        existing_public_key_user = user_public_key is not None and user_public_key.get('username') != username
-
-        match user, username_exists, password_exists, existing_public_key_user:
-            case _, True, _, True:
-                # Public key exists and doesn't match the provided username and password
-                raise Exception(f"Using user credentials in .env. User with public key {public_key} already exists but doesn't match the provided username and password. Please use a different private key (or set blank in the .env file to randomly generate with launch.sh).")
-
-            case _, False, False, True:
-                # Public key exists and no username/password provided
-                raise Exception(f"Using private key in .env. User with public key {public_key} already exists. Cannot create new user. Please use a different private key (or set blank in the .env file to randomly generate with launch.sh).")
-
-            case None, False, _, False:
+        username = os.getenv("HUB_USER")
+        password = os.getenv("HUB_PASS")
+        private_key = os.getenv("PRIVATE_KEY")
+        user = None
+        username_exist = False
+
+        public_key = get_public_key(private_key)
+
+        if username:
+            logger.info(f"Checking if user exists... User: {username}")
+            user = await hub.get_user_by_username(username)
+            username_exist = True if user else False
+        
+        if public_key and not username_exist:
+            logger.info(f"Checking if user with public key exists {public_key}...")
+            user = await hub.get_user_by_public_key(public_key)
+        
+        if user:
```

Comment:
Any reason that you changed from using match and case here? I find the nested if statements harder to follow?

Reply:
Nothing specific, I changed it to have better readability and also easier case separation

Reply:
Do you find it easier to read with the nested if statements? I find it more difficult.. If we want to change this, we should create an issue on GitHub for discussing which everyone finds better. 

The current PR should only make changes to address storing the key in a separate file.